File: OCH_Tao/ABC/B/227.js
const S = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")[1].split(" ").map(x => parseInt(x, 10));
let X = new Set();
for (let a = 1; a < 1001; a++) {
  for (let b = 1; b < 1001; b++) {
    X.add(4 * a * b + 3 * a + 3 * b);
  }
}
console.log(S.filter(x => !X.has(x)).length);

File: OCH_Tao/ABC/B/227.py
N = int(input())
S = list(map(int,input().split()))
X = set()
for a in range(1,1001):
  for b in range(1,1001):
    X.add(4*a*b+3*a+3*b)
print(len([i for i in S if i not in X]))